sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java

```diff
@@ -157,6 +156,7 @@ public static Date addDays(Date date, int numberOfDays) {
 
     ThreadSafeDateFormat(String format) {
       this.format = format;
+      getDateFormat().setLenient(false);
```

Comment:
I don't like to call getDateFormat() here to set the lenient property. If the DateFormat is not get by using this method it won't work.
I think the lenient property should be placed inside the get() method : 
```
          SimpleDateFormat simpleDateFormat = new SimpleDateFormat(format);
          simpleDateFormat.setLenient(false);
          softRef = new SoftReference<DateFormat>(simpleDateFormat);
```